src/utils/Assets.ts

```diff
@@ -0,0 +1,191 @@
+import { SHA256 } from 'crypto-js'
+import { ethers, Signer } from 'ethers'
+import {
+  Aquarius,
+  DatatokenCreateParams,
+  Nft,
+  NftCreateData,
+  NftFactory,
+  ProviderInstance,
+  ZERO_ADDRESS,
+  // approveWei,
+  // ProviderComputeInitialize,
+  // ConsumeMarketFee,
+  // Datatoken,
+  // Config,
+  // DDO,
+  // ProviderFees,
+  getEventFromTx,
+  DispenserCreationParams,
+  FreCreationParams,
+  ConfigHelper
+} from '../../src'
+import { hexlify } from 'ethers/lib/utils'
+import { createHash } from 'crypto'
+
+const VALID_TEMPLATE_INDEXES = [1, 2, 4]
+/**
+ *
+ * @param name asset name
+ * @param symbol asse symbol
+ * @param owner owner address
+ * @param assetUrl asset url
+ * @param templateIndex 1,2 or 4
+ * @param ddo ddo
+ * @param encryptDDO encrypt or not?
+ * @param providerUrl the provider URL
+ * @param providerFeeToken the provider fee token
+ * @param nftContractAddress the nft contract address
+ * @param aquariusInstance aquarius, could be node instance url
+ * @param dispenserAddress dispenser address
+ * @param fixedRateAddress fixed rate exchange address
+ * @param baseTokenAddress base token address (ocean)
+ * @returns ddo id as string
+ */
+export async function createAsset(
+  name: string,
+  symbol: string,
+  owner: Signer,
+  assetUrl: any,
+  templateIndex: number,
+  ddo: any,
+  encryptDDO: boolean = true, // default is true
+  providerUrl: string,
+  providerFeeToken: string,
+  nftContractAddress: string, // addresses.ERC721Factory,
+  aquariusInstance: Aquarius,
+  filesObject?: any
+  // fixed rate
+  // dispenserAddress?: string,
+  // fixedRateAddress?: string,
+  // baseTokenAddress?: string // ocean token?
+): Promise<string> {
+  if (!VALID_TEMPLATE_INDEXES.includes(templateIndex)) {
+    throw new Error('Invalid template index: ' + templateIndex)
+  }
+
+  const chainID = (await owner.provider.getNetwork()).chainId
+
+  const config = new ConfigHelper().getConfig(parseInt(String(chainID)))
+
+  const nft = new Nft(owner, chainID)
+
+  const nftFactory = new NftFactory(nftContractAddress, owner)
+
+  // get nft owner
+  const account = await owner.getAddress()
+
+  // from hex to number format
+  ddo.chainId = parseInt(chainID.toString(10))
+  const nftParamsAsset: NftCreateData = {
+    name,
+    symbol,
+    templateIndex,
+    tokenURI: 'aaa',
+    transferable: true,
+    owner: account
+  }
+  const datatokenParams: DatatokenCreateParams = {
+    templateIndex,
+    cap: '100000',
+    feeAmount: '0',
+    paymentCollector: account,
+    feeToken: providerFeeToken || config.oceanTokenAddress,
+    minter: account,
+    mpFeeAddress: ZERO_ADDRESS
+  }
+
+  // include fileObject in the DT constructor
+  if (templateIndex === 4) {
```

Comment:
if network is confidential EVM.
We need to extend config with this extra key.

confidential EVM = Sapphire, Sapphire testnet.   In the future we will have more, like Ocean's L1 or OceanEnterprise chains